test/e2e/rebalance_test.go

```diff
@@ -0,0 +1,214 @@
+		_, err = f.KubeClient.CoreV1().PersistentVolumeClaims(f.Ns).Update(ctx, pvc, metav1.UpdateOptions{})
+		return err
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Wait for the automated rebalancing to quick in
+	err = framework.Poll(time.Second, 5*time.Minute, func() error {
+		balanced, err := r.WorkloadCorrectlyBalanced(ctx, workload)
+		if err != nil {
+			return err
+		}
+		if balanced {
+			return nil
+		}
+		return fmt.Errorf("Expected workload %s/%s to be correctly balanced between multiple nodes", workload.Namespace, workload.Name)
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Verify that the PVC isn't annotated anymore
+	pvc, err := f.KubeClient.CoreV1().PersistentVolumeClaims(f.Ns).Get(ctx, "prometheus-k8s-db-prometheus-k8s-0", metav1.GetOptions{})
+	if _, found := pvc.Annotations[rebalancer.DropPVCAnnotation]; found {
+		t.Fatalf("Expected %s annotation to have been removed from the PVC %s/%s", rebalancer.DropPVCAnnotation, f.Ns, "prometheus-k8s-db-prometheus-k8s-0")
+	}
+
+	// Verify that no node was left unschedulable by CMO
+	nodeList, err := f.KubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{LabelSelector: "!node-role.kubernetes.io/master"})
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, node := range nodeList.Items {
+		_, cordonedByOperator := node.Annotations[rebalancer.CordonAnnotation]
+		if node.Spec.Unschedulable && cordonedByOperator {
+			t.Fatalf("Expected node %s cordoned by CMO to have been uncordoned after rebalancing the pods", node.Name)
+		}
+	}
+}
+
+func incorrectlyRebalanceWorkload(ctx context.Context, r *rebalancer.Rebalancer, workload *rebalancer.Workload) error {
+	// Cordon all nodes except one
+	nodeList, err := f.KubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{LabelSelector: "!node-role.kubernetes.io/master"})
+	if err != nil {
+		return err
+	}
+	drainer := &drain.Helper{Ctx: ctx, Client: f.KubeClient}
+	for _, node := range nodeList.Items[1:] {
+		err = drain.RunCordonOrUncordon(drainer, &node, cmodrain.Cordon)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Uncordon all the nodes once the pods are scheduled on the same node
+	defer func() {
+		nodeList, err = f.KubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{LabelSelector: "!node-role.kubernetes.io/master"})
+		for _, node := range nodeList.Items {
+			drain.RunCordonOrUncordon(drainer, &node, cmodrain.Uncordon)
+		}
+	}()
+
+	// Reschedule prometheus-k8s pods to the only schedulable node
+	err = f.KubeClient.CoreV1().PersistentVolumeClaims(f.Ns).DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: "app.kubernetes.io/name=prometheus"})
+	if err != nil {
+		return err
+	}
+	err = f.KubeClient.CoreV1().Pods(f.Ns).DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: "app.kubernetes.io/name=prometheus"})
+	if err != nil {
+		return err
+	}
+
+	// Make sure that both replicas of prometheus-k8s are scheduled on the same node
```

Comment:
couldn't we check that CMO reports the Upgradeable condition as false? This would require less code here and validate that the operator is reporting the correct condition.